### PR TITLE
scale buildkit to zero

### PR DESCRIPTION
**Pull request by wpjunior**

### CODE REVIEW

File: pkg/build/gc/gc.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		}
+
+		statefulset, err := clientset.AppsV1().StatefulSets(ns).Get(ctx, buildkitStefulset, v1.GetOptions{})
+
+		if err != nil {
+			klog.Errorf("failed to get statefullsets from ns: %s, err: %s", ns, err.Error())
+			continue
+		}
+
+		statefulset.Spec.Replicas = &zero
```
Comment (wpjunior) [inline pkg/build/gc/gc.go]: TODO: @gvicentin we may save the current replicas into a annotation

File: pkg/build/buildkit/scaler/downscaler.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			continue
+		}
+
+		if maxEndtimeByNS[pod.Namespace] < maxUsage {
+			maxEndtimeByNS[pod.Namespace] = maxUsage
+		}
+	}
+
+	now := time.Now().Unix()
+	gracefulPeriod := int64(60 * 60 * 2) // 2 Hours
```
Comment (gvicentin) [inline pkg/build/buildkit/scaler/downscaler.go]: Can we have this as an argument parameter, such as `BuildKitAutoDiscoveryGracefulPeriod`?

Comment (wpjunior) [reply pkg/build/buildkit/scaler/downscaler.go]: Yeap, we can!

File: main.go [hunk 3]
```diff
@@ -40,6 +40,7 @@ var cfg struct {
 	BuildKitAutoDiscoveryKubernetesPodSelector                string
 	BuildKitAutoDiscoveryKubernetesNamespace                  string
 	BuildKitAutoDiscoveryKubernetesLeasePrefix                string
+	BuildKitAutoDiscoveryScaleStatefulset                     string
```
Comment (gvicentin) [inline main.go]: Just a suggestion, but should we call this `BuildKitAutoDiscoveryKubernetesStatefulset` following the previous pattern?

**Pull request merged**